packages/react-components/react-nav-preview/library/src/components/SplitNavItem/useSplitNavItem.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { getIntrinsicElementProps, slot } from '@fluentui/react-utilities';
+import type { SplitNavItemProps, SplitNavItemState } from './SplitNavItem.types';
+import { useNavContext_unstable } from '../NavContext';
+import { Button, MenuButton, ToggleButton } from '@fluentui/react-button';
+import { MoreHorizontalFilled, Pin20Regular } from '@fluentui/react-icons';
+import { NavItem } from '../NavItem/index';
+
+/**
+ * Create the state required to render SplitNavItem.
+ *
+ * The returned state can be modified with hooks such as useSplitNavItemStyles_unstable,
+ * before being passed to renderSplitNavItem_unstable.
+ *
+ * @param props - props from this instance of SplitNavItem
+ * @param ref - reference to root HTMLDivElement of SplitNavItem
+ */
+export const useSplitNavItem_unstable = (
+  props: SplitNavItemProps,
+  ref: React.Ref<HTMLDivElement>,
+): SplitNavItemState => {
+  const { navItem, actionButton, toggleButton, menuButton, children } = props;
+
+  const { size = 'medium' } = useNavContext_unstable();
+
+  const navItemShorthand = slot.optional(navItem, {
+    defaultProps: {
+      children,
+    },
+    elementType: NavItem,
```

Comment:
renderByDefault: true would be suitable for this one I think, it might be good to have 'renderByDefault: false' on the others also to show optional intent

Reply:
added